### ISSUE TITLE
env XDG_SESSION_TYPE=tty is incorrect after xfce start

### ISSUE THREAD

**Issue by fish47** [fairyglade/ly#241]
the direct problem is plank's complain
```
[CRITICAL 11:48:30.374562] [AbstractMain:257] Only X11 environments are supported.
```

a simple (dirty?) solution is to force set XDG_SESSION_TYPE env, like this.
```c
setenv("XDG_SESSION_TYPE", "x11", 1);
```

any better solution?

**Comment by wolfhechel** [comment]
I get similar issues with current master, I use ly with .xinitrc that boots dwm.
I've traced it down to ly not passing metadata to the pam_systemd module. Because of this the module then sets XDG_SESSION_TYPE to tty, since PAM_XDISPLAY isn't set either.

The fix is to update env_xdg to use pam_setenv and do it right after the pam_start() call.

Strangely, this issue has already been detected and mended in https://github.com/nullgemm/ly/pull/173 but it doesn't seem to have been accepted.

@nullgemm any reason why this was closed without merging?
